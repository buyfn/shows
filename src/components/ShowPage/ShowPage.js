import React from 'react';

import EpisodeList from '../EpisodeList';

import './ShowPage.scss';

class ShowPage extends React.Component {
  componentDidMount() {
    const showID = this.props.match.params.showID;
    this.props.fetchShow(showID);
  }

  render() {
    const {
      image,
      name,
      summary,
      episodes,
      fetchingState,
      errorMsg
    } = this.props;

    if (fetchingState === 'fetching') {
      return 'Fetching data...';
    }

    if (fetchingState === 'failed') {
      return errorMsg;
    }

    return (
      <main className="show">
        <div className="show__info">
          {image && (
            <div className="show__cover">
              <img
                className="show__image"
                alt={`"${name}" show cover`}
                src={image.original}
              />
            </div>
          )}

          <div className="show__description">
            <h1 className="show__title">{name}</h1>
            <div className="show__summary" dangerouslySetInnerHTML={{ __html: summary }} />
          </div>
        </div>

        <EpisodeList episodes={episodes} />
      </main>
    );
  }
}

export default ShowPage;
