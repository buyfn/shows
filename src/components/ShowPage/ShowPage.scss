.show__info {
    display: flex;
    flex-direction: column;

    @media (min-width: 768px) {
        align-items: stretch;
        flex-direction: row;
    }
}

.show__cover {
    display: flex;
    justify-content: center;

    @media (min-width: 768px) {
        width: 50%;
        align-items: center;
    }
}

.show__image {
    max-width: 100%;
    max-height: 100%;
    object-fit: cover;
}

.show__description {
    display: flex;
    flex-direction: column;
    padding: 2rem;

    @media (min-width: 768px) {
        width: 50%;
        justify-content: center;
    }
}
