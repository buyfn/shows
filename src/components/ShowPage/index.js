import { connect } from 'react-redux';
import { fetchShowRequest } from '../../actions';

import ShowPage from './ShowPage';

const mapStateToProps = (state) => ({
  ...state.show,
  fetchingState: state.fetchingState,
  errorMsg: state.errorMsg,
});

const mapDispatchToProps = dispatch => ({
  fetchShow: (id) => dispatch(fetchShowRequest(id)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ShowPage);
