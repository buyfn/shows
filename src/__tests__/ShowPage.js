import React from 'react';
import renderer from 'react-test-renderer';

import ShowPage from '../components/ShowPage/ShowPage';

describe('ShowPage', () => {
  const props = {
    match: {
      params: { showID: '1' }
    },
    summary: 'test show summary',
    name: 'test show name',
    image: {
      medium: 'img.jpg',
      original: 'img.jpg'
    },
  };

  let tree;
  const fetchShowMock = jest.fn();

  beforeAll(() => {
    tree = renderer
      .create(<ShowPage {...props} fetchShow={fetchShowMock} />)
      .toJSON();
  });

  it('calls "fetchShow" with correct argument', () => {
    expect(fetchShowMock).toHaveBeenCalledWith('1');
    expect(fetchShowMock.mock.calls.length).toEqual(1);
  });

  it('renders correctlly', () => {
    expect(tree).toMatchSnapshot();
  });
});