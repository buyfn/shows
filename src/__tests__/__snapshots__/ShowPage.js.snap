// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowPage renders correctlly 1`] = `
<main
  className="show"
>
  <div
    className="show__info"
  >
    <div
      className="show__cover"
    >
      <img
        alt="\\"test show name\\" show cover"
        className="show__image"
        src="img.jpg"
      />
    </div>
    <div
      className="show__description"
    >
      <h1
        className="show__title"
      >
        test show name
      </h1>
      <div
        className="show__summary"
        dangerouslySetInnerHTML={
          Object {
            "__html": "test show summary",
          }
        }
      />
    </div>
  </div>
</main>
`;
